elasticsearch/security_api.go

```diff
@@ -0,0 +1,220 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
```

Comment:
not yet, but i was intending to test most of it as integration tests. what do you think should be tested as unit test?

Reply:
There is some parameter handling and setting, decoding of responses, error handling etc going on here. IMO this should be tested in unit tests. 
Since we now maintain a _fat_ ES client, there should be dedicated integration tests with ES for this package. I am fine with covering with system tests for the specific use cases for now, but really suggest to at least create an issue for adding package integration tests.

Reply:
With the changed logic I agree that unit tests are not required. When I wrote the comment a lot of request body encoding, response decoding and handling, closing response body etc was going on.